### DISCUSSION TITLE
OJS in offline environment

### DISCUSSION THREAD

**Discussion by linogaliana** [quarto-dev/quarto-cli#5971] [Q&A] [answered]
### Description

I created a few weeks ago an app using OJS cells with `quarto` that can be found [here](https://linogaliana.github.io/serverless-annotation-tool/) (source code [here](https://github.com/linogaliana/serverless-annotation-tool)).

I wanted to run it locally (using `quarto_preview`) in a network that has no access to internet. When trying to run it (using `quarto preview` I discovered that an internet connection was needed for some libraries like `D3`, `inputs`. I tried to add some requirements based on copies of the relevant libraries in my local filesystem. I tried to reference them using the following syntax: 

```javascript
d3 = require(await FileAttachment('/js/d3.min.js').url())
inputs = require(await FileAttachment('/js/inputs.min.js').url())
marked = require(await FileAttachment('/js/marked.min.js').url())
htl = require(await FileAttachment('/js/htl.min.js').url())
```

But I still have the `unable to load module` error in my document. It looks like the local libraries are loaded :

![image](https://github.com/quarto-dev/quarto-cli/assets/33896139/0a5ce0b4-6e58-43b8-9999-83df99eb820b)

But the error persists. Do you know if additional manual configurations are needed in offline environments ?

**Comment by cscheid** [comment, marked as answer]
There's too many parts of the OJS environment that require an active internet connection. We unfortunately don't support this usage.

**Reply by neocarto** [reply]
It's a real shame. In some countries, we need to be able to work without an Internet connection.

**Reply by cscheid** [reply]
I agree, but investigate the source code of observable's JS framework yourself and you'll come to the same conclusion about the amount of changes we'd have to make.
